File: HowTheyWrite_android/app/src/main/java/com/kaiserpudding/howtheywrite/lessonList/LessonListActivity.java
package com.kaiserpudding.howtheywrite.lessonList;

import android.arch.lifecycle.ViewModelProviders;
import android.content.Intent;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import com.kaiserpudding.howtheywrite.R;
import com.kaiserpudding.howtheywrite.model.Lesson;

public class LessonListActivity extends AppCompatActivity {

  private static final int NEW_LESSON_ACTIVITY_REQUEST_CODE = 1;

  private LessonListViewModel lessonListViewModel;

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_lesson_list);

    RecyclerView recyclerView = findViewById(R.id.lesson_recyclerview);
    final LessonListAdapter adapter = new LessonListAdapter(this);
    recyclerView.setAdapter(adapter);
    recyclerView.setLayoutManager(new LinearLayoutManager(this));

    lessonListViewModel = ViewModelProviders.of(this).get(LessonListViewModel.class);
    lessonListViewModel.getLessons().observe(this, adapter::setLessons);

    FloatingActionButton fab = findViewById(R.id.add_lessen_fab);
    fab.setOnClickListener(v -> {
      Intent intent = new Intent(LessonListActivity.this, NewLessonActivity.class);
      startActivityForResult(intent, NEW_LESSON_ACTIVITY_REQUEST_CODE);
    });
  }

  @Override
  public void onActivityResult(int requestCode, int resultCode, Intent data) {
    super.onActivityResult(requestCode, resultCode, data);

    if (requestCode == NEW_LESSON_ACTIVITY_REQUEST_CODE && resultCode == RESULT_OK) {
      Lesson lesson = new Lesson(data.getStringExtra(NewLessonActivity.EXTRA_REPLY));
      lessonListViewModel.insertLesson(lesson);
    }
  }

}

File: HowTheyWrite_android/app/src/main/java/com/kaiserpudding/howtheywrite/lessonList/LessonListActivity.kt
package com.kaiserpudding.howtheywrite.lessonList

import android.app.Activity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import android.content.Intent
import com.google.android.material.floatingactionbutton.FloatingActionButton
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.kaiserpudding.howtheywrite.R
import com.kaiserpudding.howtheywrite.model.Lesson

class LessonListActivity : AppCompatActivity() {

    private var lessonListViewModel: LessonListViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_lesson_list)

        val recyclerView = findViewById<RecyclerView>(R.id.lesson_recyclerview)
        val adapter = LessonListAdapter(this)
        recyclerView.adapter = adapter
        recyclerView.layoutManager = LinearLayoutManager(this)

        lessonListViewModel = ViewModelProviders.of(this).get(LessonListViewModel::class.java)
        lessonListViewModel!!.lessons!!.observe(this, Observer<List<Lesson>> { adapter.setLessons(it!!) })

        val fab = findViewById<FloatingActionButton>(R.id.add_lessen_fab)
        fab.setOnClickListener {
            val intent = Intent(this@LessonListActivity, NewLessonActivity::class.java)
            startActivityForResult(intent, NEW_LESSON_ACTIVITY_REQUEST_CODE)
        }
    }

    public override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
        super.onActivityResult(requestCode, resultCode, data)

        if (requestCode == NEW_LESSON_ACTIVITY_REQUEST_CODE && resultCode == Activity.RESULT_OK) {
            val lesson = Lesson(data!!.getStringExtra(NewLessonActivity.EXTRA_REPLY))
            lessonListViewModel!!.insertLesson(lesson)
        }
    }

    companion object {

        private val NEW_LESSON_ACTIVITY_REQUEST_CODE = 1
    }

}
